src/Generator.h

```diff
@@ -328,6 +330,137 @@ template <typename T> class GeneratorParam : public Internal::GeneratorParamBase
     }
 };
 
+/** Addition between GeneratorParam and arithmetic/enum types.
+ * Returns arithmetic type. */
```

Comment:
These comments now seem incorrect. How about "GeneratorParam<T> acts like T for the purpose of all operator overloads".